src/coreclr/utilcode/stresslog.cpp

```diff
@@ -185,13 +188,19 @@ static LPVOID CreateMemoryMappedFile(LPWSTR logFilename, size_t maxBytesTotal)
     }
 
     size_t fileSize = maxBytesTotal;
+    // TODO: PAL does not support mapping larger than 32 bit
     HandleHolder hMap = WszCreateFileMapping(hFile, NULL, PAGE_READWRITE, (DWORD)(fileSize >> 32), (DWORD)fileSize, NULL);
     if (hMap == NULL)
     {
         return nullptr;
     }
 
+#ifdef HOST_WINDOWS
     return MapViewOfFileEx(hMap, FILE_MAP_ALL_ACCESS, 0, 0, fileSize, (void*)0x400000000000);
+#else
+    // TODO: PAL does not support mapping with a specified base address
+    return MapViewOfFileEx(hMap, FILE_MAP_ALL_ACCESS, 0, 0, fileSize, nullptr);
```

Comment:
While we possibly might be able to add support for this, it would not make sense to pick any static address on Unix. Moreover, it would be just a hint anyways. So let's do it the way you did, you can remove the TODO from this.
A nit - it would be nicer to make a `#define` for the base address that would be platform specific and leave the code the same.

Reply:
With some experimentation, I believe `StressLogAnalyzer` works by mapping the file back to exactly where it was. The reason is that once it map the file back to memory, it started tracing pointers, and these pointers could be pointing into the mapped area. For example, we are dereferencing `log.t` here:

https://github.com/dotnet/runtime/blob/8fbf206d0e518b45ca855832e8bfb391afa85972/src/coreclr/tools/StressLogAnalyzer/StressLogPlugin/StressLogPlugin.cpp#L1145

I wonder if we could make it work by sending the base address parameter to the first parameter of `mmap`, they seem to serve the same purpose.

I understand there is a possibility for the API to fail, it looks like I have to handle the relocation of those pointers regardless :(

Reply:
The first parameter to mmap is just a hint. The mmap will completely ignore it when some SELinux security measures are on and it would also pick a different address if there already is a mapping overlapping the range. I guess that in Windows address space, we can rely on this address range being always (?) available, but we don't have that luxury on Unix. So it would be better to not to rely on a specific base address.